test/src/org/omegat/core/data/AutoTmxTest.java

```diff
@@ -128,6 +132,66 @@ public boolean existEntryInProject(EntryKey key) {
         checkTranslation(ste, "bizbaz", TMXEntry.ExternalLinked.xENFORCED);
     }
 
+    /**
+     * Test that an alternate translation matching an id is correctly applied
+     * when the memory is in tm/enforce.
+     */
+    @Test
+    public void test_alternateOnEnforce() throws Exception {
+        ExternalTMX enforceTMX = prepareTestTM();
+
+        SourceTextEntry ste = createSTE("1_0", "Edit");
+        p.allProjectEntries.add(ste);
+        checkTranslation(ste, "foobar", null);
+        p.importHandler = new ImportFromAutoTMX(p, p.allProjectEntries);
+        p.appendFromAutoTMX(enforceTMX, true);
+        checkTranslation(ste, "alternative", TMXEntry.ExternalLinked.xENFORCED);
+    }
+
+    /**
+     * Test that an alternate translation matching an id is correctly applied
+     * when the memory is in tm/auto.
+     */
+    @Test
+    public void test_alternateOnAuto() throws Exception {
+        ExternalTMX enforceTMX = prepareTestTM();
+
+        SourceTextEntry ste = createSTE("1_0", "Edit");
+        p.allProjectEntries.add(ste);
+        checkTranslation(ste, "foobar", null);
+        p.importHandler = new ImportFromAutoTMX(p, p.allProjectEntries);
+        p.appendFromAutoTMX(enforceTMX, false);
+        checkTranslation(ste, "alternative", TMXEntry.ExternalLinked.xAUTO);
+    }
```

Comment:
Just a suggestion and it's probably overkill but since you have set up everything already maybe add a test_alternateInDefaultTmFolder to check that nothing happens when the tmx is added to the default folder?
I don't think test1 and test2 do that...?
And while I'm suggesting things if you feel like renaming them, I think that would be great :)

Reply:
>  to check that nothing happens when the tmx is added to the default folder

Unfortunately, this doesn't seem to be easy to test as the field `org.omegat.core.data.RealProject.transMemories` and the method `org.omegat.core.data.RealProject.loadTM()` are private. I'm not really sure this PR is right for refactoring this class.

> And while I'm suggesting things if you feel like renaming them, I think that would be great 

yes, thanks for the suggestion, it's done in e21efec67d0f03c496be327e327aa8840d19d27b.